core/src/main/java/org/jclouds/rest/internal/RestAnnotationProcessor.java

```diff
@@ -779,6 +782,22 @@ private static void addHeader(Multimap<String, String> headers, Headers header,
       return parts.build();
    }
 
+   private static GeneratedHttpRequest stripExpectHeaderIfContentZero(GeneratedHttpRequest request) {
+      boolean isBodyEmpty = false;
+      if (request.getPayload() != null) {
+         Long length = request.getPayload().getContentMetadata().getContentLength();
+         if (length != null && length == 0) {
+            isBodyEmpty = true;
+         }
+      } else {
+         isBodyEmpty = true;
+      }
```

Comment:
This could be a bit cleaner:
```java
boolean isBodyEmpty = true;
if (request.getPayload() != null) {
   Long length = request.getPayload().getContentMetadata().getContentLength();
   isBodyEmpty = (length == null || length == 0);
}
```